src/documents/to-do/DeveloperIntroductionToInfusionFramework.md

```diff
@@ -4,6 +4,595 @@ layout: default
+ * [`fluid.viewComponent`](/infusion/development/ComponentConfigurationOptions.md#view-components), which supplement model components with functionality to bind to the DOM of a web page.
+
+Component-based development emphasizes declarative configuration, loose coupling and flexible hierarchies.
+
+A new Infusion component is defined using the `fluid.default` function and a very basic "Hello, World!" component might look like this:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/oBYvwx?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+
+```
+// The first argument is the name of the new component, the second is the
+// new component's default configuration
+fluid.defaults("fluid.helloWorld", {
+    // gradeNames supplies an array of strings that name previously
+    // defined Infusion components
+    gradeNames: ["fluid.component"]
+});
+```
+
+After definition, instances of the component can be created by calling the component's name as a function, with the option to pass in further configuration options as an argument:
+
+```
+var helloWorld = fluid.helloWorld({});
+```
+
+Right now this component doesn't do anything new, but we will evolve its definition throughout this introduction to demonstrate further core concepts of the framework.
+
+### Invokers
+
+All Infusion components support the definition of public functions using [invokers](/infusion/development/Invokers.md); among other things, invokers allow related components to have different internal implementations of a function but present a common function to be called by other code. If we imagine a "Hello, World!" component, there are a variety of ways it might say "hello". Two of them are:
+
+* By printing to the web developer console via `console.log`
+* By displaying a message on a web page
+
+We'll define the first approach as an invoker on the "Hello, World!" component. Other styles of invokers are possible, but we'll use one here that lets us refer to functions of existing Javascript objects:
+
+<div class="infusion-docs-note">You can check out the [Live Example](http://codepen.io/waharnum/pen/MJbgEx?editors=1111) of the code below on [CodePen](http://codepen.io/)</div>
+
+```
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        sayHello: {
+            // Configures this invoker to use the console object
+            "this": "console",
+            // Configures this invoker to use the log function of the console
+            // object
+            "method": "log",
+            // Configures the arguments to pass to the method
+            "args": ["Hello, World!"]
+        }
+    }
+});
+```
+
+Now we can create an instance of the component and call the invoker as a function to say hello:
+
+```
+var helloWorld = fluid.helloWorld({});
+helloWorld.sayHello();
+```
```

Comment:
At the risk of making this too complex, IMO it would be good to demonstrate how an invoker can be overridden.